server/service/osquery.go

```diff
@@ -818,14 +818,34 @@ func (svc *Service) SubmitDistributedQueryResults(
 				ll = level.Error(svc.logger)
 			}
 			ll.Log("query", query, "message", messages[query], "hostID", host.ID)
-
 		}
+
+		// live queries we do want to ingest even if the query had issues, because we want to inform the user of these
+		// issues
+		// same applies to policies, since it's a 3 state result, one of them being failure, and labels take this state
+		// into account as well
+
 		var err error
+		switch {
+		case strings.HasPrefix(query, hostDistributedQueryPrefix):
+			err = svc.ingestDistributedQuery(ctx, *host, query, rows, failed, messages[query])
+		case strings.HasPrefix(query, hostPolicyQueryPrefix):
+			err = ingestMembershipQuery(hostPolicyQueryPrefix, query, rows, policyResults, failed)
+		case strings.HasPrefix(query, hostLabelQueryPrefix):
+			err = ingestMembershipQuery(hostLabelQueryPrefix, query, rows, labelResults, failed)
+		}
+		logIngestionError(ctx, err)
```

Comment:
You are calling this twice for some queries now. Maybe that's a-ok?
Or try to merge the two `switches`? (the ones in line 829 and 844)

Reply:
Hm, yeah, good point. I tied that err to failed, but it's a separate thing... will rethink it...